table.go

```diff
@@ -69,17 +65,25 @@ func NumDigits(b *big.Int) int64 {
 	if bl == 0 {
 		return 1
 	}
-	if val, ok := lookupBits(bl); ok {
-		ab := new(big.Int).Abs(b)
-		if ab.Cmp(&val.border) < 0 {
+
+	if bl < digitsTableLen {
+		val := digitsLookupTable[bl]
+		a := b
+		if b.Sign() < 0 {
+			a = new(big.Int).Abs(b)
+		}
+		if a.Cmp(&val.border) < 0 {
 			return val.digits
 		}
 		return val.digits + 1
 	}
 
 	n := int64(float64(bl) / digitsToBitsRatio)
-	a := new(big.Int).Abs(b)
 	e := new(big.Int).Exp(bigTen, big.NewInt(n), nil)
```

Comment:
Yes, but the lookup table won't ever be used there because we only hit this branch if the lookup table for this function missed. In benchmarks using tableExp10 is 0 faster. I've switched to it to be consistent, but it's not a win.